### PR TITLE
Adding environment variable support to BuildRequest

### PR DESCRIPTION
**Pull request by jupierce**
In order to support feature request: https://trello.com/c/Yl4qomsw/878-5-additional-jenkins-plugin-enhancement-requests-from-github

Adding environment variable support to restclient on which Jenkins plugin relies.

fyi @gabemontero

### CODE REVIEW

File: src/main/java/com/openshift/internal/restclient/model/build/BuildRequest.java [hunk 1]
```diff
@@ -36,7 +35,10 @@
 	private static final String REVISION_TYPE = REVISION + "." + TYPE;
 	private static final String TRIGGERED_BY = "triggeredBy";
 	private static final String MESSAGE = "message";
-	
+	private static final String ENV = "env";
+	private static final String ENV_NAME = "name";
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/build/BuildRequest.java]: NAME and VALUE are available as part of the interface implemented by KubernetesResource

**Comment by jcantrill** [comment]
[test]  LGTM.  @jupierce Test for BuildRequest to validate its set properly?

File: src/main/java/com/openshift/internal/restclient/model/build/BuildRequest.java (hunk 1 above)
Comment (jupierce) [inline src/main/java/com/openshift/internal/restclient/model/build/BuildRequest.java]: Fixed in latest push. Thanks.

**Comment by jupierce** [comment]
@jcantrill Added BuildRequestTest.java unit test.

**Comment by jcantrill** [comment]
[merge]

**Comment by gabemontero** [comment]
cool - thanks @jupierce and @jcantrill

**Pull request merged**